client/lib/deploy.rs

```diff
@@ -273,9 +272,23 @@ impl DeployExt for Deploy {
             transfer_args.insert(TRANSFER_ARG_SOURCE, source_purse)?;
         }
 
-        let target_account_hash = target_account.to_account_hash().value();
-        transfer_args.insert(TRANSFER_ARG_TARGET, target_account_hash)?;
+        if let Ok(public_key) = PublicKey::from_hex(target_account) {
+            transfer_args.insert(TRANSFER_ARG_TARGET, public_key)?;
+        } else if let Ok(account_hash) = AccountHash::from_formatted_str(target_account) {
+            transfer_args.insert(TRANSFER_ARG_TARGET, account_hash)?;
+        } else if let Ok(uref) = URef::from_formatted_str(target_account) {
+            transfer_args.insert(TRANSFER_ARG_TARGET, uref)?;
+        } else {
+            return Err(Error::InvalidArgument {
+                context: "target_account",
+                error: format!(
+                    "Could not parse value as a PublicKey, AccountHash or URef: {}",
```

Comment:
```suggestion
                    "Allowed types: PublicKey, AccountHash or URef, got {}",
```
I would just make the error message more surgical/sterile.